### PR TITLE
add hooks and tests for findOneAndRemove and update

### LINKED ISSUE #6: enhancement: provide more middleware hooks
Mongoose supports middleware for these functions now:

- `findOneAndRemove`
- `findOneAndUpdate`
- `update` 

So far _historical_ is only hooked to `save`.

### PR DESCRIPTION
**Pull request by kevinlmadison**
Added the rest of the hooks and tests relevant to #6 , I believe this should actually close that issue now unless we want even more hooks. These were pretty straight forward after taking care of findOneAndUpdate.

closes #6

### CODE REVIEW

**Comment by kevinlmadison** [comment]
I think I know what the issue is. The way the test works for the update hook is can happen out of order it seems, so I need to force it to wait.

**Comment by stennettm** [comment]
Fantastic. I really appreciate the help here, thanks.

**Comment by kevinlmadison** [comment]
No problem, you did all the heavy lifting creating this in the first place so thank you for that.

File: historical.js [hunk 1]
```diff
@@ -205,12 +205,55 @@ module.exports = function (schema, options) {
         });
     });
 
-    schema.pre('update', function (next) {
-        next();
+    schema.post('update', function (next) {
+        var update = this.getUpdate().$set,
+            pathing = getPaths(this.getUpdate());
+
+        this.model.findOne(update).exec().then(function(doc) {
```
Comment (stennettm) [inline historical.js]: Update can affect more than one document, but we're archiving the changes of only one.

**Pull request merged**